### DISCUSSION TITLE
Examples with TGN or cGNNs training on many small graphs and parallelisations + documentation of examples

### DISCUSSION THREAD

**Discussion by SalvishGoomanee** [pyg-team/pytorch_geometric#9034] [Ideas]
Hello, 

I am working with on a number of variants of the TGN or cGNNs and was wondering if submitting pull requests pertaining to examples of variants of such models trainings on small datasets (custom or QM9) and also on the parallelisation of the training procedures (with either `cuda` or `mps`)? 

I was, also, wondering if adding documentation to the scripts in `/examples` could have any help and if yes, I would be happy to do it!

**Comment by rusty1s** [comment]
We started to add more documentation to `examples` (e.g., [here](https://github.com/pyg-team/pytorch_geometric/tree/master/examples/multi_gpu)), but the top-level README still needs work. Happy to get help on this one.

Regarding your question: Can you clarify what features you plan to add?

**Reply by SalvishGoomanee** [reply]
Sure, I can start on this. 

For the TGN, a small example that shows how to include a node_pos encoder for treating the pos features in order to have a show how to train a more geometrically principled temporal network (treating rot + trans. inv/equiv.)? 

But then I, also, don't see EGNNs models (eg: as in [here](https://arxiv.org/pdf/2102.09844.pdf) now). This could be interesting/simpler to add first?

**Reply by rusty1s** [reply]
I would personally like to avoid making examples to complex. In general, all examples model a certain research paper or highlight a specific PyG feature. If you want to add a TGN example with node position encoder, do you think this warrant a completely new example? Is there a specific paper you have in mind that uses this technique?

**Reply by SalvishGoomanee** [reply]
I am not aware of any paper yet but I have in mind the generalisation of this example [here](https://github.com/chaitjo/geometric-gnn-dojo/blob/main/geometric_gnn_101.ipynb), but yes for sure, it will bit more involved. 
I can share a gist.
